src/api/proxy/CacheMixin.ts

```diff
@@ -0,0 +1,113 @@
+import NodeCache from 'node-cache'
+
+interface CacheOptions<T> {
+  method: keyof T
+  ttl?: number
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  hashFn?: (...params: any[]) => string
+}
+
+interface CacheMethodParams<T, P, R> extends CacheOptions<T> {
+  fnToCache: (params: P) => R
+  instance: T
+}
+
+export class CacheMixin {
+  private cache: NodeCache
+
+  public constructor() {
+    this.cache = new NodeCache({ useClones: false })
+  }
+
+  /**
+   * Injects into provided `instance` a version with cache according to the list `toCache`.
+   * Injected methods will obey `ttl` if provided, otherwise global cache config.
+   *
+   * @param instance Class instance to cache
+   * @param toCache List of configs for methods to be cached
+   */
+  public injectCache<T>(instance: T, toCache: CacheOptions<T>[]): void {
+    toCache.forEach(cacheConfig => {
+      const { method } = cacheConfig
+      const methodName = method.toString()
```

Comment:
Isn't method already a string, or TS complains?

Reply:
No, method is a type. TS does complain.